include/eve/algo/any_of.hpp

```diff
@@ -7,37 +7,57 @@
 //==================================================================================================
 #pragma once
 
-#include <eve/algo/preprocess_range.hpp>
 #include <eve/algo/for_each_iteration.hpp>
+#include <eve/algo/preprocess_range.hpp>
+#include <eve/algo/traits.hpp>
 
 #include <eve/function/any.hpp>
+#include <eve/function/logical_or.hpp>
 
-
+#include <array>
 
 namespace eve::algo
 {
   struct any_of_
   {
-    struct default_traits {};
+    static constexpr traits default_traits{unroll<4>};
 
     template <typename P>
     struct delegate
     {
       explicit delegate(P p) : p(p) {}
 
-      bool step(auto it, eve::relative_conditional_expr auto ignore)
+      EVE_FORCEINLINE bool step(auto it, eve::relative_conditional_expr auto ignore)
       {
-        auto test = p(eve::load[ignore](it));
+        eve::logical test = p(eve::load[ignore](it));
         res = eve::any[ignore](test);
         return res;
       }
 
+      template <typename I, std::size_t size>
+      EVE_FORCEINLINE bool unrolled_step(std::array<I, size> arr)
+      {
+        // TODO: Issue #710 to do the proper reduciton;
+        auto f = arr.begin(), l = arr.end();
```

Comment:
eve/detail/meta.hpp had a for_ that helps unrolling, maybe use it here ?

Reply:
Didn't touch this.
This one is a very temp one, we'll fix it in a little bit.